twisted/twisted/plugins/tunnel_helper_plugin.py

```diff
@@ -130,9 +162,7 @@ def start_tribler(self):
         config.set_enable_torrent_search(False)
         config.set_enable_channel_search(False)
         self.session = Session(config)
-        upgrader = self.session.prestart()
-        while not upgrader.is_done:
-            time.sleep(0.1)
+        self.session.prestart()
```

Comment:
You should still check whether the upgrader failed or not. I did something like this:

```
upgrader = self.session.prestart()
if upgrader.failed:
    msg("The upgrader failed: .Tribler directory backed up, aborting")
    reactor.addSystemEventTrigger('after', 'shutdown', os._exit, 1)
    reactor.stop()
else:
    self.session.start()
    msg("Tribler started")
```